pandas/core/indexes/base.py

```diff
@@ -353,7 +353,7 @@ def __new__(
             return Index(data, dtype=object, copy=copy, name=name, **kwargs)
 
         # index-like
-        elif isinstance(data, (np.ndarray, Index, ABCSeries)):
+        elif isinstance(data, (np.ndarray, ABCIndexClass, ABCSeries)):
```

Comment:
Can you limit scope of change in this module to just `ABCExtensionArray`? I think that's the only one that can be exclusively replaced with the ABC... import

The rest let's leave separate